### ISSUE TITLE
[Package Request]: SciTE4AutoIt

### ISSUE THREAD

**Issue by jiaojiaodubai** [microsoft/winget-pkgs#88875]
### Package Requested

- [X] I would like someone else to build the manifest.
- [ ] I would like help so I can submit the manifest.
- [X] I have performed a search and couldn't find this package.
- [ ] I have checked that the installer for this package is not contained in a .zip file
- [ ] I think there is a new version available and I have provided the URL.

### Please provide the following information

Publisher: AutoIt Team
Publisher Url: https://www.autoitscript.com/site/autoit-script-editor/downloads/
Package Name:SciTE4AutoIt
Description: Special edition of SciTE customized by AutoIt community for AutoIt scripting language.
Package Version: 21.316.1639.0
Installer URL: https://www.autoitscript.com/cgi-bin/getfile.pl?../autoit3/scite/download/SciTE4AutoIt3.exe

Current Package Identifier (for new versions):

**Comment by borque** [comment]
Hello!
This seems difficult due to the server configuration at autoitscript.com:
The download-url you provided replies with "403 - Forbidden" when used for creating a winget-manifest. 
It points to https://www.autoitscript.com/autoit3/scite/download/SciTE4AutoIt3.exe whicht shows the same behaviour.
Both URLs work perfectly normal when used in a webbrowser, i.e. they start the download.
I was unable to find any other (direct) download-url on major download portals, they all only provide links to the publisher url you provided where the download has to be initiated manually (by clicking on the link).
**tl;dr**
autoitscript.com seems to accept downloads only from "real" browsers not from download-tools. I have no idea how to create a manifest for this unless either you find another (trustworthy) download-url or autoitscript.com changes the download policy or winget can mimic Firefox or Edge or ... .

**Comment by jiaojiaodubai** [comment]
> Hello! This seems difficult due to the server configuration at autoitscript.com: The download-url you provided replies with "403 - Forbidden" when used for creating a winget-manifest. It points to https://www.autoitscript.com/autoit3/scite/download/SciTE4AutoIt3.exe whicht shows the same behaviour. Both URLs work perfectly normal when used in a webbrowser, i.e. they start the download. I was unable to find any other (direct) download-url on major download portals, they all only provide links to the publisher url you provided where the download has to be initiated manually (by clicking on the link). **tl;dr** autoitscript.com seems to accept downloads only from "real" browsers not from download-tools. I have no idea how to create a manifest for this unless either you find another (trustworthy) download-url or autoitscript.com changes the download policy or winget can mimic Firefox or Edge or ... .

Anyway, thank you very much for your efforts!
I was surprised because we could find and download AutoIt with winget, but we couldn't download its dedicated editor. Although this is not the only option for editing AutoIt script files, it seems that other editors (such as VS Code) need to rely on some of its files (AutoIt3Wrapper. au3) to work properly.
I tried to use the search engine to find other trusted download links of SciTE4AutoIt, but I got nothing useful:  ( I will mention this in the AutoIt forum, and hope we can solve this problem.

**Comment by borque** [comment]
If you find a solution for the download problem I can try to assist by builing the manifest.

**Comment by borque** [comment]
Hello!
Just a short answer: the link used by Trenly in PR #72385 is https://www.autoitscript.com/files/autoit3/autoit-v3-setup.exe while yours pointed to https://www.autoitscript.com/autoit3/scite/download/SciTE4AutoIt3.exe for me. You should notice the different directories.
I am quiet busy "with real life" for the rest of the week, but unless the AutoIt-Forum provides a better solution at the weekend I can try it with "Trenlys link" modified for ScITe.

**Comment by borque** [comment]
Another short report: trying to modify Tenlys https://www.autoitscript.com/files/autoit3/SciTE4AutoIt3.exe , https://www.autoitscript.com/files/scite/SciTE4AutoIt3.exe or https://www.autoitscript.com/files/autoit3/scite/SciTE4AutoIt3.exe does not work. 404 - Not found.

**Comment by borque** [comment]
Meanwhile I was able to solve the download problem. For some reason the URL does not work when used in wingetcreate, but can be used in YamlCreate without problems.
Unfortunately after downloading another major obstacle occurs: SciTE4AutoIt requires a working installation of AutoIt3 to be installable.Unless package dependencies are implemented by winget (see here: microsoft/winget-cli#163 ) the manifest can't be added.
I have already prepared manifest files that should meet the (future) requirements and will append them here for reference (txt needs to be changed to yaml to get them working). But unless package dependencies are implemented we should add something like "Blocking-Issue" here.

`
[AutoIt.SciTE4AutoIt.installer.txt](https://github.com/microsoft/winget-pkgs/files/10098235/AutoIt.SciTE4AutoIt.installer.txt)
`

`
[AutoIt.SciTE4AutoIt.locale.en-US.txt](https://github.com/microsoft/winget-pkgs/files/10098236/AutoIt.SciTE4AutoIt.locale.en-US.txt)
`

`
[AutoIt.SciTE4AutoIt.txt](https://github.com/microsoft/winget-pkgs/files/10098238/AutoIt.SciTE4AutoIt.txt)
`

**Comment by borque** [comment]
@MSFTbot Blocking-Issue

**Comment by borque** [comment]
@MSFTbot Dependencies

**Comment by borque** [comment]
@stephengillie , @denelon , @mdanish-kh  : As a normal user I don't have the right to add labels to this request. If you agree with my analysis could one of you do so please?

**Comment by mdanish-kh** [comment]
@msftbot Dependencies

**Comment by mdanish-kh** [comment]
@msftbot Blocking-Issue

**Comment by jiaojiaodubai** [comment]
> Meanwhile I was able to solve the download problem. For some reason the URL does not work when used in wingetcreate, but can be used in YamlCreate without problems. Unfortunately after downloading another major obstacle occurs: SciTE4AutoIt requires a working installation of AutoIt3 to be installable.Unless package dependencies are implemented by winget (see here: [microsoft/winget-cli#163](https://github.com/microsoft/winget-cli/issues/163) ) the manifest can't be added. I have already prepared manifest files that should meet the (future) requirements and will append them here for reference (txt needs to be changed to yaml to get them working). But unless package dependencies are implemented we should add something like "Blocking-Issue" here or close the issue.
> 
> `[AutoIt.SciTE4AutoIt.installer.txt](https://github.com/microsoft/winget-pkgs/files/10098235/AutoIt.SciTE4AutoIt.installer.txt)`
> 
> `[AutoIt.SciTE4AutoIt.locale.en-US.txt](https://github.com/microsoft/winget-pkgs/files/10098236/AutoIt.SciTE4AutoIt.locale.en-US.txt)`
> 
> `[AutoIt.SciTE4AutoIt.txt](https://github.com/microsoft/winget-pkgs/files/10098238/AutoIt.SciTE4AutoIt.txt)`

Thank you for keeping track of this issue.

Yes, SciTE4AutoIt is an editor specially designed for AutoIt3. Under normal conditions, after installing AutoIt3, it contains a cut down version of SciTE4AutoIt. After that, for better editing experience, we will install the full version of SciTE4AutoIt. The full version of SciTE4AutoIt will be automatically installed to the installation location of AutoIt3, and the original cut down version will be overwritten. Perhaps during the installation of SciTE4AutoIt, it will read the registry to the installation location of AutoIt3. If this location does not exist, the installation of SciTE4AutoIt will be interrupted.

In addition, I also bring a piece of good news. The developers of SciTE4AutoIt have personally tried to add SciTE4AutoIt to Winget, reproducing the "403" problem you encountered. He has informed the website administrator of AutoIt3 about this situation, and soon the administrator will deal with the problem that SciTE4AutoIt.exe is difficult to download.

**Comment by borque** [comment]
Hello @jiaojiaodubai !
This is good news indeed. As I have also filed an issue-report for winget-create here https://github.com/microsoft/winget-create/issues/326 because the download-url is working in YamlCreate.ps1 and browsers but not in winget-create it would be good if the AutoIt admins (or you) could give a short report (under the above issue-repot) what caused the problem when they have fixed it.
Background: if they fix it without reporting, the devs of winget-create won't be able to reproduce the bug and will close the issue without being able to fix it.
